test/test_dependencies.py

```diff
@@ -189,3 +189,43 @@ def test_colcon_defaults(tmpdir):
     assert 'ros-dashing-ament-cmake' not in result
     assert 'ros-dashing-rclcpp' not in result
     assert 'ros-dashing-rclpy' in result
+
+
+@uses_docker
+def test_dummy_skip_rosdep_keys_doesnt_exist_pkg(tmpdir):
+    ws = Path(str(tmpdir))
+    pkg_xml = Path(ws) / 'src' / 'dummy' / 'package.xml'
+    pkg_xml.parent.mkdir(parents=True)
+    pkg_xml.write_text(CUSTOM_KEY_PKG_XML)
+    client = DockerClient()
+    platform = Platform(arch='aarch64', os_name='ubuntu', ros_distro='dashing')
+    out_script = ws / rosdep_install_script(platform)
+
+    skip_keys = ['definitely_does_not_exist']
+    gather_rosdeps(client, platform, workspace=ws, skip_rosdep_keys=skip_keys)
+    result = out_script.read_text().splitlines()
+    expected = [
+        '#!/bin/bash',
+        'set -euxo pipefail',
```

Comment:
Is this the only thing we need to check?
Is there a specific string we're supposed to be looking for? I myself am not sure how to better test this...

Reply:
Well the idea is to test if those specific lines which includes the keys are not there in the script. But did it other way.